vowpalwabbit/CMakeLists.txt

```diff
@@ -345,9 +352,9 @@ add_library(vw STATIC ${vw_all_sources} ${vw_all_headers})
 
 target_link_libraries(vw
   PUBLIC
-    VowpalWabbit::explore VowpalWabbit::allreduce Boost::boost
+    VowpalWabbit::explore VowpalWabbit::allreduce Boost::boost vw_io
```

Comment:
I think we should leave vw_io as a private dep but add spdlog and fmt here as public deps